docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,608 @@
+  which prioritizes the packaged listed there in the solves. For example: if you started your
+  environment with `conda create -n something python=3.9 numpy=1.21`, `numpy` will be _soft-pinned_
+  during the lifetime of the environment. Soft-pinning means that the solver will try to respect
+  that specific version (1.21) in every solve, unless the user asks for a different version
+  explicitly (thus overriding the historic preference). For napari, this means that every plugin
+  installation will be recorded in the history file, accumulating over time. If we compound this
+  with `napari` updates, the problem gets larger with every new release.
+* Guarantee of success: updating an environment to the latest napari release might now work right
+  away, specially if the user has installed plugins that have conflicting packaging metadata. Even
+  if the installation succeeds, insufficient/incorrect metadata might result in the wrong packages
+  being installed, rendering the napari installation non-operational!
+* Using several versions side-by-side: the multiple environments setup fits nicely in contexts
+  where several napari versions (or even "flavors") are needed.
+
+Since we install plugins to the napari environments, we need to guarantee that the new version
+will be compatible with the installed plugins. To do so, we instruct `conda` to perform a _dry-run_
+install. If the environment can be solved, we can proceed with no interruptions. If the environment
+is not solvable, then we can offer the user two options:
+
+* Start a fresh environment only containing napari, with no extra plugins, keeping the old version
+  in place.
+* (Experimental idea / suggestion) Run a potentially time-consuming analysis on which package(s)
+  are creating the installation conflicts and suggest which plugins cannot be installed. This kind
+  of analysis would entail a series of dry runs, removing one plugin at a time, hoping that the
+  environment gets solved eventually.
+
+It must be considered that better metadata at napari hub could substantially improve the
+installability analysis for the end-user. If we keep track of which napari versions the plugins are
+compatible with (either by running some kind of CI ourselves or making this analysis part of the
+submission procedure), we could simply query the API to anticipate which packages are installable
+before running the update.
+
+#### Tasks
+
+- [ ] Detect availability of new napari versions
+- [ ] Create new environment with only napari
+- [ ] Migrate plugins over to the new environment
+- [ ] Implement "co-installability" analysis
+
+#### Risks
+
+Co-installability of plugins is ultimately a matter of metadata and good practices. The risks here
+are similar to the concerns shared in Milestone 1.
+
+### Milestone 5: Deprecating Briefcase-based installers
+
+Once we are satisfied with how `conda` installers are working for the end user, we can think of
```

Comment:
My hot take:

The bundled app has always been considered experimental/alpha/unsupported etc AFAIK. If we're committed to the conda-based approach and this NAP, then we should drop the briefcase-based installer ASAP so the dev community doesn't feel the pressure to support two implementations.

Reply:
As long as the new bundle will continue to support installing from PyPI, I don't mind dropping the briefcase one anytime

Reply:
as I understand bundle support install from PyPi only using console?

Reply:
The conda-bundle UI doesn't support PyPI packages yet, that's true. I'd say that's a milestone we need to support (safely) in the conda-bundles before we can deprecate Briefcase-bundles. Agreed!